### PR TITLE
Use `[[maybe_unused]]` instead of `(void)` cast

### PR DESCRIPTION
**Pull request by mkghadban**
Replaced paranthesis with double brackets

### CODE REVIEW

File: source/base/function_lib.cc [hunk 1]
```diff
@@ -2606,7 +2606,7 @@ namespace Functions
     const Point<dim>  &p,
     const unsigned int component) const
```
Comment (tamiko) [inline source/base/function_lib.cc:2607]: You need to add the `[[maybe_unused]]` attribute to the function argument

File: source/base/function_lib.cc [hunk 2]
```diff
@@ -2606,7 +2606,7 @@ namespace Functions
...
   {
-    (void)component;
```
Comment (tamiko) [inline source/base/function_lib.cc:2609]: and simply remove the `(void)component` line.

**Review by tamiko — CHANGES_REQUESTED** [review]

File: source/base/function_lib.cc (hunk 2 above)
Comment (mkghadban) [reply source/base/function_lib.cc:2609]: Oh ...Thanks!

**Comment by mkghadban** [comment]
Hopefully fixed now!

**Review by masterleinad — APPROVED** [review]

**Review by tamiko — APPROVED** [review]

**Pull request merged**